### PR TITLE
Move Linux python test into docker 

### PR DESCRIPTION
**Pull request by mszhanyi**
### Description
supplement of #17417

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml [hunk 1]
```diff
@@ -141,77 +141,43 @@ stages:
...
+            --volume $(Build.SourcesDirectory):/onnxruntime_src \
+            --volume $(Build.BinariesDirectory):/build \
+            --volume /data/models:/build/models:ro \
+            --volume $HOME/.onnx:/home/onnxruntimedev/.onnx \
+            -e ALLOW_RELEASED_ONNX_OPSET_ONLY=0 \
+            -e NIGHTLY_BUILD \
+            -e BUILD_BUILDNUMBER \
+            onnxruntimecpubuild \
+              /bin/bash -c "
+                ln -s /opt/python/cp38-cp38/bin/python3 /usr/bin/python3 && \
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml:177]: The file /usr/bin/python3 may already exists. Please try to do it without using the `ln` command. You can use a full path or set up PATH env.

Comment (mszhanyi) [reply tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml:177]: No, it doesn't.
we could find there're many `/opt/python/cp38-cp38/bin/python3 `or /`opt/python/cp39-cp39/bin/python3` in the code.
I think the reason is that the image is to build cp38~cp311 wheels, so what does the python3 mean in the image, python3.8, python3.9 or others?
And it's a string, so bash parameter expansion doesn't work.
Alias or Env variable don't work in /bin/bash -c "".

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml:177]: CentOS 8 comes with python 3.6. Replacing it with python 3.8 could have cause unexpected behavior.

File: tools/scripts/python_test.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+set -ex
```
Comment (snnn) [inline tools/scripts/python_test.sh:1]: Please add "#!/bin/bash" there.

**Review by snnn — APPROVED** [review]

**Pull request merged**